sys/cddl/contrib/opensolaris/uts/common/fs/zfs/zfs_vnops.c

```diff
@@ -953,6 +953,7 @@ zfs_write(vnode_t *vp, uio_t *uio, int ioflag, cred_t *cr, caller_context_t *ct)
 			break;
 		}
 
+#ifdef illumos
```

Comment:
This is dead code upstream as the only place that sets `xuio` to a non-NULL value is under an #ifdef, and that particular code sets `iov_base` to a kernel address that code in these regions checks for via assertions, etc.  This is probably going away upstream due to ZoF anyway, so this seemed the simpler local diff to carry in the meantime.